crates/runtime/src/embeddings/udtf.rs

```diff
@@ -464,9 +465,18 @@ impl TableProvider for VectorSearchUDTFProvider {
                 false,
             )],
             input: Arc::new(proj),
-            fetch: limit,
+            // i.e., the order limit passed down via the UDTF invocation
+            fetch: self.args.limit,
```

Comment:
We could do something like this instead https://github.com/spiceai/spiceai/blob/d4683b76e36acab79e65379bc721fa7eecd2ecc8/crates/runtime/src/embeddings/index/query_table.rs#L335

```rust
 self.limit_to_use(limit),
 ```

Reply:
Sure, taking the smaller limit makes sense. Thanks!

Reply:
Do we need the subsequent `LogicalPlan::Limit` if we have the correct limit applied in the SORT?

Reply:
Good point! Sorted!